### DISCUSSION TITLE
AVR-DU

### DISCUSSION THREAD

**Discussion by MCUdude** [SpenceKonde/DxCore#131] [Future development discussion]
It doesn't seem like you're running out of work right away!

Introducing AVR-DU; with native USB!

https://ww1.microchip.com/downloads/en/DeviceDoc/AVR-DU-Product-Brief-DS40002328.pdf

**Comment by SpenceKonde** [comment]
Did you manage to save it before it vanished? It's not there anymore. 

Thank fucking god we will finally have something better than micro! 

But I think it's going to be a long way off. We don't even have DD yet!

**Comment by MCUdude** [comment]
The URL doesn't work for you?
Here's the PDF!

[AVR-DU-Product-Brief-DS40002328.pdf](https://github.com/SpenceKonde/DxCore/files/7041383/AVR-DU-Product-Brief-DS40002328.pdf)

**Comment by SpenceKonde** [comment]
Nope, thanks! 

Interesting.... what's with their stepping back down a tier in flash and ram with newer parts? They did 128k for DA and DB, but now we're only seeing 64 for DD, DU, and EA....

**Comment by SpenceKonde** [comment]
I'm getting a sinking feeling that the 14 and 20 pin AVR DD-series parts aren;'t going to be released in 64k versions :-(

**Comment by SpenceKonde** [comment]
Converted this to a discussion - the AVR DU-series is probably not going to see ther light of day for a year the way things have been going, and until more information is released, it's not an actionable issue.

**Reply by SpenceKonde** [reply]
Especially seeing as Microchip appears to have pulled the document, lol....

I just hope they provide a relatively thorough set of native USB app notes..., AVR deserves to finally get to come out and play, with native USB, modern AVR peripherals, but if the USB programming is too challenging, it will remain out of reach for most.

**Comment by chipperdoodles** [comment]
It can't just be me thinking the DU would make a good usb UDPI + Serial Port combo programmer. I'm thinking of that + potential charge controller but yeah now that they're actually out in the wild is anyone working on something like this? RN i'm just using dual channel wch342F IC for my boards which gets the job done but i'm still eyeballing this as a thought

**Reply by MCUdude** [reply]
I know @askn37 is working on a UPDI programmer based on the AVR-DU. It uses the JTAG3/EDBG protocol (same as the Curiosity Nano boards do), and it will be compatible with Avrdude. I'm not sure it will feature a virtual serial port as well though.

**Reply by askn37** [reply]
> I know @askn37 is working on a UPDI programmer based on the AVR-DU. It uses the JTAG3/EDBG protocol (same as the Curiosity Nano boards do), and it will be compatible with Avrdude. I'm not sure it will feature a virtual serial port as well though.

The UPDI4AVR-USB programmer is already equipped with a VCP (Virtual-Communication-Port from CDC-ACM) with a throughput of 500kbps to 1Mbps and, in addition to UPDI native, can also link to DTR to start the arduino/urclock bootloader, so all operations expected from the Arduino-IDE console are covered.

<img src="https://askn37.github.io/product/UPDI4AVR/images/IMG_3832.jpg" width="400"> <img src="https://askn37.github.io/product/UPDI4AVR/images/U4AU_UPDI.drawio.svg" width="400">

It can also operate TPI devices, making it compatible with the "Curiosity Nano" and "Xplained Mini/Nano" series. (If you supply power via GPIO instead of VCC, you can also operate the power supply with `-xvtarg_switch=<0,1>`.)

<img src="https://askn37.github.io/product/UPDI4AVR/images/IMG_3839.jpg" width="400"> <img src="https://askn37.github.io/product/UPDI4AVR/images/U4AU_TPI.drawio.svg" width="400">

Open testing of the software/firmware for "AVR64DU32 Curiosity Nano" will start soon. We are currently preparing the necessary documentation.

The AVR-DU series has only one analog comparator, which the UPDI4AVR uses for feedback of the high-voltage charge pump, so there are currently no plans to support secondary battery charging.

**Reply by chipperdoodles** [reply]
This is fantastic work! thank you for sharing, I'll have to share this with some friends that are also using avr chips!
